### PR TITLE
Update samples to work with TypeScript strict mode (part 1)

### PR DESCRIPTION
**Pull request by nash1111**
Like this, I want to add types to all samples to enhance safety. If there are no issues, I would like to proceed with adding them to all.

### CODE REVIEW

File: src/components/SampleLayout.tsx [hunk 1]
```diff
@@ -136,6 +136,9 @@ const SampleLayout: React.FunctionComponent<
     };
     try {
       const canvas = canvasRef.current;
+      if (!canvas) {
```
Comment (kainino0x) [inline src/components/SampleLayout.tsx:139]: An assert helper will help a lot with cases like this and make the new checks less invasive.

You can use
```
export function assert(condition: unknown, msg?: string): asserts condition {
  if (!condition) {
    throw new Error(msg);
  }
}
```

Probably easiest to just put this in SampleLayout.tsx.

File: src/sample/animometer/main.ts [hunk 2]
```diff
@@ -272,13 +282,13 @@ const init: SampleInit = async ({ canvas, pageState, gui }) => {
       }
     }
 
-    let startTime = undefined;
+    let startTime: number | undefined = undefined;
     const uniformTime = new Float32Array([0]);
 
     const renderPassDescriptor: GPURenderPassDescriptor = {
```
Comment (kainino0x) [inline src/sample/animometer/main.ts]: Committed suggestion to improve the typescript overhead here slightly.

File: src/sample/animometer/main.ts [hunk 3]
```diff
@@ -4,6 +4,12 @@ import animometerWGSL from './animometer.wgsl';
 
 const init: SampleInit = async ({ canvas, pageState, gui }) => {
   const adapter = await navigator.gpu.requestAdapter();
+  if (!adapter) {
+    console.error(
+      'WebGPU is not supported. Make sure you are running the latest version of a compatible browser (like Chrome Canary) with the correct flags enabled.'
+    );
+    return;
```
[the remark this answers is not in the document]
Comment (kainino0x) [reply src/sample/animometer/main.ts]: (Done in the commit I pushed)

File: src/components/SampleLayout.tsx (hunk 1 above)
Comment (kainino0x) [reply src/components/SampleLayout.tsx:139]: (Done in the commit I pushed)

**Comment by kainino0x** [comment]
Looking good so far, hopefully the suggestions will make updating the rest of the samples a bit easier - ping me when you need another review!

**Comment by kainino0x** [comment]
And do be sure to pull the branch before you continue, because I committed suggestions to it.

**Comment by nash1111** [comment]
Thank you for the review! I just did a git pull for your changes. Once I've corrected all the examples, I'll request a re-review from you!

**Comment by nash1111** [comment]
I am capable of handling it, but given the volume, I'm considering splitting it into two PRs. The first PR will modify only the files that have been completed so far with strict: false, and the second will edit the remaining files to make the entire project strict: true. I believe that smaller PRs are preferable, so I'll be aiming for that. This PR is the first one. 
I will submit the second PR by the end of this month.

**Review by kainino0x — APPROVED** [review]
Sounds good! LGTM

**Pull request merged**